x-pack/auditbeat/module/system/package/package.go

```diff
@@ -147,6 +149,16 @@ func (pkg Package) toMapStr() common.MapStr {
 	return mapstr
 }
 
+// entityID creates an ID that uniquely identifies this package across machines.
+func (pkg Package) entityID(hostID string) string {
+	h := system.NewEntityHash()
+	binary.Write(h, binary.LittleEndian, []byte(hostID))
+	binary.Write(h, binary.LittleEndian, []byte(pkg.Name))
+	binary.Write(h, binary.LittleEndian, []byte(pkg.Version))
+	binary.Write(h, binary.LittleEndian, int64(pkg.InstallTime.Nanosecond()))
```

Comment:
Note to self: Should not include `InstallTime` for the same reasons as outlined in https://github.com/elastic/beats/pull/10508#issue-249880577.

Also, should different versions of a package be counted as two entities?

Reply:
Removed `InstallTime`. Leaving `version` as part of the `entity_id`, so two versions of a package will count as two. I think this makes sense because there are cases where two versions are actually installed - e.g. Python 2.6 and 3.x - and they should both be counted.